src/Entities/Entity.cpp

```diff
@@ -133,9 +133,9 @@ const char * cEntity::GetParentClass(void) const
 
 
 
-bool cEntity::Initialize(cWorld & a_World)
+bool cEntity::Initialize(std::unique_ptr<cEntity> a_Entity, cWorld & a_EntityWorld)
```

Comment:
Rename `a_Entity` to `a_Self` to make it more obvious what the param should be. Also, ugly!